lib/DBDish/Pg/StatementHandle.pm6

```diff
@@ -17,7 +18,33 @@ method !handle-errors {
     if $!result.is-ok {
         self.reset-err;
     } else {
-        self!set-err($!result.PQresultStatus, $!pg_conn.PQerrorMessage);
+        self!error-dispatch: X::DBDish::DBError::Pg.new(
```

Comment:
It's entirely a matter of personal taste, but seeing as nearly all the attributes are derived from the ```$!result``` I'd be tempted to just pass that to the constructor and de-structure it in the ```BUILD``` or ```TWEAK```` as appropriate.